tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,380 @@
+"""BleBox cover entities tests."""
+
+import blebox_uniapi
+import pytest
+
+from homeassistant.components.cover import (
+    ATTR_CURRENT_POSITION,
+    ATTR_POSITION,
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GATE,
+    DEVICE_CLASS_SHUTTER,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+    SUPPORT_CLOSE,
+    SUPPORT_OPEN,
+    SUPPORT_SET_POSITION,
+    SUPPORT_STOP,
+)
+from homeassistant.const import (
+    ATTR_DEVICE_CLASS,
+    ATTR_SUPPORTED_FEATURES,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER,
+    STATE_UNKNOWN,
+)
+
+from .conftest import async_setup_entity, mock_feature
+
+from tests.async_mock import ANY, AsyncMock, PropertyMock, call, patch
+
+ALL_COVER_FIXTURES = ["gatecontroller", "shutterbox", "gatebox"]
+
+
+@pytest.fixture(name="shutterbox")
+def shutterbox_fixture():
+    """Return a shutterBox fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-shutterBox-2bee34e750b8-position",
+        full_name="shutterBox-position",
+        device_class="shutter",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My shutter")
+    type(product).model = PropertyMock(return_value="shutterBox")
+    return (feature, "cover.shutterbox_position")
+
+
+@pytest.fixture(name="gatebox")
+def gatebox_fixture():
+    """Return a gateBox fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateBox-1afe34db9437-position",
+        device_class="gatebox",
+        full_name="gateBox-position",
+        current=None,
+        state=None,
+        has_stop=False,
+        is_slider=False,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gatebox")
+    type(product).model = PropertyMock(return_value="gateBox")
+    return (feature, "cover.gatebox_position")
+
+
+@pytest.fixture(name="gatecontroller")
+def gate_fixture():
+    """Return a gateController fixture."""
+    feature = mock_feature(
+        "covers",
+        blebox_uniapi.cover.Cover,
+        unique_id="BleBox-gateController-2bee34e750b8-position",
+        full_name="gateController-position",
+        device_class="gate",
+        current=None,
+        state=None,
+        has_stop=True,
+        is_slider=True,
+    )
+    product = feature.product
+    type(product).name = PropertyMock(return_value="My gate controller")
+    type(product).model = PropertyMock(return_value="gateController")
+    return (feature, "cover.gatecontroller_position")
+
+
+async def test_init_gatecontroller(gatecontroller, hass, config):
+    """Test gateController default state."""
+
+    feature_mock, entity_id = gatecontroller
+    entity = await async_setup_entity(hass, config, entity_id)
```

Comment:
Please rename `entity` to `entry` or similar as it's an entity registry entry and not an entity instance.